### PR TITLE
Add leader election and traefik as a dependency

### PR DESCRIPTION
**Pull request by maulikjs**
https://issues.redhat.com/browse/RHODS-1394

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ verify_ssl = true
 openshift = "==0.11.2"
 jupyterhub-singleuser-profiles = "==0.4.2"
 oauthenticator = {ref = "d98936da37569b7bb400c6854b07041dce3e7f87",git = "https://github.com/opendatahub-io/oauthenticator.git", editable = true}
+jupyterhub-traefik-proxy = {ref = "b468da04bf0057e9baeb458de6c3f9f677cfb79e", git = "https://github.com/opendatahub-io/traefik-proxy.git", editable = true}
```
Comment (anishasthana) [inline Pipfile:12]: Shouldn't this refer to rhds traefik proxy?

Comment (maulikjs) [reply Pipfile:12]: Depends, so its exactly the same code as upstream, in which case shouldn't we be using upstream libraries? 🤔

Comment (anishasthana) [reply Pipfile:12]: I'm leaning towards it anyways. @crobby  / @tmckayus  whaddya think?

Comment (tmckayus) [reply Pipfile:12]: I was just thinking about this. We should probably fork it, we have with everything else.

Comment (tmckayus) [reply Pipfile:12]: One thing to note here, though -- CPaaS doesn't use Pipfile. It uses the requirements*.txt

So this change will have no effect for a CPaaS build, only an upstream build of this repo. To pick it up in CPaaS, we need the requirements.txt altered.

Comment (maulikjs) [reply Pipfile:12]: ack yeah I did not know that
Ill update that file!

Comment (maulikjs) [reply Pipfile:12]: Updated

File: requirements-external.txt [hunk 2]
```diff
@@ -7,3 +7,5 @@ git+https://github.com/opendatahub-io/oauthenticator@42c4790a2db77ff4259a74e15ed
 # jupyterhub-singleuser-profiles/jupyterhub_singleuser_profiles/ui/package-lock.json
 #
 git+https://github.com/red-hat-data-services/jupyterhub-singleuser-profiles@d6c088f96a95a92c9aa8aa2a3fe657552ad82071#egg=jupyterhub-singleuser-profiles
+git+https://github.com/opendatahub-io/traefik-proxy@b468da04bf0057e9baeb458de6c3f9f677cfb79e#edd=jupyterhub-traefik-proxy
```
Comment (tmckayus) [inline requirements-external.txt:10]: typo here, "edd" instead of "egg"

otherwise lgtm

Comment (maulikjs) [reply requirements-external.txt:10]: ah shoot lol

Comment (maulikjs) [reply requirements-external.txt:10]: updated

**Review by anishasthana — APPROVED** [review]
/lgtm

**Pull request merged**